src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalMessagesOpt.java

```diff
@@ -62,11 +62,14 @@ public void getComponentMessages(MessageCacheItem cacheItem) {
 				path = Paths.get(uri);
 				messages = JSONBundleUtil.getMessages(path);
 			}
-			cacheItem.setCacheItem(messages, null, System.currentTimeMillis(), null);
 		} catch (Exception e) {
 			logger.debug(e.getMessage());
 			// Do not update cacheItem
 		}
+		if (messages!=null && !messages.isEmpty())
+			cacheItem.setCacheItem(messages, null, System.currentTimeMillis(), null);
+		else
+			new LocalSrcMessagesPropsOpt(this.dto).getComponentMessages(cacheItem);
```

Comment:
What's the logic behind the code? It looks not good to have this kind of logic.

Reply:
Do you mean what is the issue? Please see the linked issue for details. This is for product teams that insist on using multiple .properties file instead of the VIP messages_source.json format.

Reply:
By the way, this PR is not ready for review. I will re-request review once ready.

Reply:
@Xiaochao8 , you do not need to approve this PR, if the solution does not look good. Do you have any suggestions? I am thinking of adding a new getMessage API with the source bundle as parameter just so we support the use case in the issue. Let me know what you think.

Reply:
I'm not very clear about the requirements and your design. But it looks not good to have this kind of logic in code.

Reply:
Sure, let's talk offline. Put this PR on hold.